### PR TITLE
pythonPackages.hypercorn: init at 0.11.2

### PR DESCRIPTION
**Pull request by dgliwka**
###### Motivation for this change

[Hypercorn](https://pgjones.gitlab.io/hypercorn/) - an ASGI Server based on Hyper libraries and inspired by Gunicorn.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after): 132.3M
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123905` at eb687747 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.hypercorn</li>
    <li>python39Packages.hypercorn</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-explicit-check-phase.md">python-explicit-check-phase</a>

Consider using `pytestCheckHook` in `checkInputs` rather than `checkPhase = "pytest";`.

Near pkgs/development/python-modules/hypercorn/default.nix:40:3:
```
   |
40 |   checkPhase = "pytest";
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/hypercorn/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  checkInputs = [
+    pytest
+    pytest-asyncio
+    pytest-cov
+    pytest-sugar
+    pytest-trio
+    ];
+
+  checkPhase = "pytest";
```
Comment (jonringer) [inline pkgs/development/python-modules/hypercorn/default.nix:40]: Would you please use `pytestCheckHook` instead? This is a bit more nix friendly, and also ensures that tests are run as part of a build.

More information can be found on the [pytestCheckHook](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook) section of the manual.

Comment (dgliwka) [reply pkgs/development/python-modules/hypercorn/default.nix]: Done, thank you.

File: pkgs/development/python-modules/hypercorn/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "0v80v6l2xqac5mgrmh2im7y23wpvz4yc2v4h9ryhvl88c2jk9mvh";
+  };
+
+  propagatedBuildInputs = [ wsproto toml h2 priority ]
+  ++ lib.optionals (pythonOlder "3.8") [ typing-extensions ];
+
+  checkInputs = [
+    pytest
```
Comment (jonringer) [inline pkgs/development/python-modules/hypercorn/default.nix:34]:
```suggestion
```

File: pkgs/development/python-modules/hypercorn/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitLab
+, pythonOlder
+, typing-extensions
+, wsproto
+, toml
+, h2
+, priority
+, pytest
```
Comment (jonringer) [inline pkgs/development/python-modules/hypercorn/default.nix:10]:
```suggestion
```

**Review by jonringer — COMMENTED** [review]
pytestCheckHook will bring in pytest for you.

otherwise LGTM

**Comment by ethancedwards8** [comment]
Result of `nixpkgs-review pr 123905` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python39Packages.hypercorn</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>python38Packages.hypercorn</li>
  </ul>
</details>

File: pkgs/development/python-modules/hypercorn/default.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  propagatedBuildInputs = [ wsproto toml h2 priority ]
+  ++ lib.optionals (pythonOlder "3.8") [ typing-extensions ];
+
+  checkInputs = [
+    pytest-asyncio
+    pytest-cov
+    pytest-sugar
+    pytest-trio
+    pytestCheckHook
+    ] ++ lib.optionals (pythonOlder "3.8") [ mock ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hypercorn/default.nix:39]:
```suggestion
  ] ++ lib.optionals (pythonOlder "3.8") [ mock ];
```

File: pkgs/development/python-modules/hypercorn/default.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+  src = fetchFromGitLab {
+    owner = "pgjones";
+    repo = pname;
+    rev = version;
+    sha256 = "0v80v6l2xqac5mgrmh2im7y23wpvz4yc2v4h9ryhvl88c2jk9mvh";
+  };
+
+  propagatedBuildInputs = [ wsproto toml h2 priority ]
+  ++ lib.optionals (pythonOlder "3.8") [ typing-extensions ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hypercorn/default.nix:31]:
```suggestion
    ++ lib.optionals (pythonOlder "3.8") [ typing-extensions ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by jonringer — APPROVED** [review]
LGTM

```
https://github.com/NixOS/nixpkgs/pull/123905

2 packages built:
python38Packages.hypercorn python39Packages.hypercorn
```

**Comment by jonringer** [comment]
@dgliwka sorry for the delay.

Congrats on your first contribution to nixpkgs :)

**Pull request merged**